src/notation/internal/notationinteraction.cpp

```diff
@@ -3616,18 +3639,55 @@ void NotationInteraction::copyLyrics()
     QApplication::clipboard()->setText(text);
 }
 
+inline QString extractSyllable(const QString& txt, int& textPos)
+{
+    textPos = txt.indexOf(QRegExp("\\S"), textPos);
+    if (textPos == -1) {
+        textPos = 0;
+        return "";
+    }
+    static QString regex = QString("[^\\S") + QChar(0xa0) + QChar(0x202F) + "]+";
+    int splitPos = txt.indexOf(QRegExp(regex), textPos);
+    if (splitPos == -1) {
+        splitPos = txt.length();
+    }
+    QString result = txt.mid(textPos, splitPos - textPos);
+    int hyphPos = result.indexOf('-');
+    if (hyphPos != -1) {
+        result = result.mid(0, ++hyphPos);
+        textPos += hyphPos;
+    } else {
+        textPos = splitPos + 1;
+    }
+    return result;
+}
+
 void NotationInteraction::pasteSelection(const Fraction& scale)
 {
     startEdit();
 
     if (isTextEditingStarted()) {
-        #if defined(Q_OS_MAC) || defined(Q_OS_WIN)
+#if defined(Q_OS_MAC) || defined(Q_OS_WIN)
         QClipboard::Mode mode = QClipboard::Clipboard;
-        #else
+#else
         QClipboard::Mode mode = QClipboard::Selection;
-        #endif
+#endif
```

Comment:
I still believe this stuff should go and only the `QClipboard::Mode mode = QClipboard::Clipboard;` should stay

Reply:
Of course, this needs to be fixed in a separate PR, not here

Reply:
I see. I'll rebase my PR then, once this here is merged ;-)

Reply:
or you could cherry-pick mine